pint/quantity.py

```diff
@@ -124,6 +124,9 @@ def __str__(self):
     def __repr__(self):
         return "<Quantity({0}, '{1}')>".format(self._magnitude, self._units)
 
+    def __hash__(self):
+        return hash(self.__class__) + hash(self.magnitude) + hash(self.units)
```

Comment:
This should probably be `hash((self.__class__, self.magnitude, self.units))`, to allow `tuple.__hash__` to combine the values in a better way that addition